golem-service-base/src/db/mod.rs

```diff
@@ -13,17 +13,32 @@
 // limitations under the License.
 
 use std::error::Error;
+use std::ops::Deref;
 use std::path::Path;
 
 use sqlx::postgres::{PgConnectOptions, PgPoolOptions};
 use sqlx::sqlite::{SqliteConnectOptions, SqlitePoolOptions};
 use sqlx::{Connection, Executor, PgConnection, Pool, Postgres, Sqlite, SqliteConnection};
 use tracing::info;
 
-use crate::config::{DbPostgresConfig, DbSqliteConfig};
+use golem_common::config::{DbPostgresConfig, DbSqliteConfig};
```

Comment:
are these wrappers useful? maybe we should just add `SqliteConnectOptions::from_config`, then we could get rid of them.

Reply:
You mean to add into `sqlx::sqlx_sqlite::SqliteConnectOptions` a method `from_config` ?

Reply:
This is what the wrapper and Deref are doing, private to this mod only.

Reply:
technically it does, i just see no point in having `DbPostgresConfigWrapper`. from this name alone one cannot find out that this is a wrapper created only to provide "config to opts" as a From instance, and this could be helped with more precise naming, i just think it does not worth it now.

when config was not extracted, then i think this was okay, but now it is just plumbing (wrapping for the purpose of unwrapping) with no good reason (at least for me).

i would simply just add `xxx_options_from_config`, or `xxx_config_to_options` for sqlite and pg

Reply:
You mean to define something like 

```
Db[Postgres|Sqlite]Config:: xxx_options_from_config -> [Pg|Sqlite]ConnectOptions
Db[Postgres|Sqlite]Config:: xxx_config_to_options -> [Pg|Sqlite]ConnectOptions 
```
In `Db[Postgres|Sqlite]Config` ?

If it is that, we need to add sqlx dependency in golem-common, which I understood we want to avoid.

Reply:
https://github.com/golemcloud/golem/pull/875
i just meant using simple functions :)